core/src/main/java/com/google/zxing/common/BitArray.java

```diff
@@ -163,15 +163,8 @@ public void setRange(int start, int end) {
     for (int i = firstInt; i <= lastInt; i++) {
       int firstBit = i > firstInt ? 0 : start & 0x1F;
       int lastBit = i < lastInt ? 31 : end & 0x1F;
-      int mask;
-      if (firstBit == 0 && lastBit == 31) {
-        mask = -1;
-      } else {
-        mask = 0;
-        for (int j = firstBit; j <= lastBit; j++) {
-          mask |= 1 << j;
-        }
-      }
+      // Ones from firstBit to lastBit, inclusive
+      int mask = (2 << lastBit) - (1 << firstBit);
```

Comment:
That's pretty cool, never though of writing it this way. Seems to work. You can inline this in the line below if you like.